src/store/modules/order/actions.ts

```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
     } catch(error) {
       showToast(translate("Something went wrong"));
     }
+    return resp;
+  },
+
+  async updateAppliedFilters({ state, commit, dispatch }, payload) {
+    if (payload.filterName === 'selectedPoIds') {
+      const poIds = state.query.selectedPoIds;
+      !poIds.includes(payload.value) ? poIds.push(payload.value) : poIds.splice(poIds.indexOf(payload.value), 1)
+      payload.value = poIds
+    }
+    commit(types.ORDER_FILTERS_UPDATED, payload)
+    const resp = await dispatch('updateQuery', { viewSize: process.env.VUE_APP_VIEW_SIZE, viewIndex: 0, queryString: state.query.queryString })
+    return resp;
+  },
+
+  async updateSortOption({ commit }, payload) {
+    commit(types.ORDER_SORT_UPDATED, payload)
+  },
+
+  async updatePoIds({ commit }, payload) {
+    commit(types.ORDER_PO_ID_UPDATED, payload)
+    return payload;
+  },
+
+  async updateQuery({ state, dispatch }, params) {
+    await dispatch('updateQueryString', params.queryString)
```

Comment:
I think we should update query in applied filters and dispatch updateQuery instead

Reply:
Done Sir